src/Components/CreateImageWizardV2/steps/Oscap/Oscap.tsx

```diff
@@ -0,0 +1,274 @@
+import React, { useEffect, useState } from 'react';
+
+import {
+  Alert,
+  FormGroup,
+  Spinner,
+  Popover,
+  TextContent,
+  Text,
+  Button,
+} from '@patternfly/react-core';
+import {
+  Select,
+  SelectOption,
+  SelectVariant,
+} from '@patternfly/react-core/deprecated';
+import { HelpIcon } from '@patternfly/react-icons';
+
+import OscapProfileInformation from './OscapProfileInformation';
+
+import { useAppDispatch, useAppSelector } from '../../../../store/hooks';
+import {
+  DistributionProfileItem,
+  useGetOscapCustomizationsQuery,
+  useGetOscapProfilesQuery,
+} from '../../../../store/imageBuilderApi';
+import {
+  changeOscapProfile,
+  selectDistribution,
+  selectProfile,
+} from '../../../../store/wizardSlice';
+
+/**
+ * Component for the user to select the profile to apply to their image.
+ * The selected profile will be stored in the `oscap-profile` form state variable.
+ * The Component is shown or not depending on the ShowSelector variable.
+ */
+
+const ProfileSelector = () => {
+  const oscapProfile = useAppSelector((state) => selectProfile(state));
+  const release = useAppSelector((state) => selectDistribution(state));
+  const dispatch = useAppDispatch();
+  const [profileName, setProfileName] = useState<string | undefined>('');
+  const [isOpen, setIsOpen] = useState(false);
+  const {
+    data: profiles,
+    isFetching,
+    isSuccess,
+    isError,
+    refetch,
+  } = useGetOscapProfilesQuery({
+    distribution: release,
+  });
+
+  const { data } = useGetOscapCustomizationsQuery(
+    {
+      distribution: release,
+      // @ts-ignore
```

Comment:
Let's sort out the rest of the comments and see if we can get rid of this ignore.